server/src/main/java/io/crate/planner/operators/LogicalPlan.java

```diff
@@ -121,7 +121,8 @@ default boolean preferShardProjections() {
      * <p>
      *  Note that `outputsToKeep` can contain scalars on top of the outputs that the "current" operator outputs.
      *  This doesn't mean that the operator has to pull-down the scalar as well, but it means it has to provide all outputs
-     *  that are required by the parent.
+     *  that are required by the parent. The new pruned outputs should be in the same original order, to avoid errors
+     *  regarding {@link Union} which requires the order to be kept, or introduction of unnecessary {@link Eval} operators.
```

Comment:
I'd also add a `@param outputsToKeep` that says that the collection must have a stable order